### PR TITLE
Add Transactions Details Dialog

### PR DESCRIPTION
**Pull request by blankey1337**
image
image
image

### CODE REVIEW

File: web/src/app/lib/i18n/i18n.service.ts [hunk 1]
```diff
@@ -51,7 +51,16 @@ export class I18nService {
...
+    return this.data && this.data[phrase] ? this.interpolate(this.data[phrase], opts) : phrase
+  }
+
+  private interpolate(phrase:string, opts?:object): string {
+    const toBeTranslated = phrase.match(/__([^ ]*)__/gm);
+    if (!toBeTranslated) return phrase;
+    for (var i = 0; i < toBeTranslated.length; i++) {
+      phrase = phrase.replace(toBeTranslated[i], opts && opts[toBeTranslated[i].replace(/__/g, "")]);
+    }
+    return phrase;
```
Comment (blankey1337) [inline web/src/app/lib/i18n/i18n.service.ts:63]: adds backwards compatibility with the `__varname__` format of the current locale files

File: web/src/app/pages/dashboard/transactions/transaction-details-dialog/transaction-details-dialog.component.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    this.dialogRef.close();
+  }
+
+  public getInfoData(): TransactionDetailsCellValueMap[] {
+    return Array.from(this.infoData.entries());
+  } 
+
+  public getDetailsData(): TransactionDetailsCellValueMap[] {
+    return Array.from(this.detailsData.entries());
+  }
```
Comment (blankey1337) [inline web/src/app/pages/dashboard/transactions/transaction-details-dialog/transaction-details-dialog.component.ts:48]: `ngFor` doesn't play nice with `Map`s yet, so we have to use `Array.from` for now.. :(

File: web/src/app/pages/dashboard/transactions/transaction-row-value-cell/transaction-row-value-cell.component.ts [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        break;
+      }
+      case EncryptedMessage: {
+        this.valueType = 'EncryptedMessage';
+        break;
+      }
+      case String:
+        if (BurstUtil.isBurstcoinAddress(this.value as string)) {
+          this.valueType = 'BurstAddress';
+        }
```
Comment (blankey1337) [inline web/src/app/pages/dashboard/transactions/transaction-row-value-cell/transaction-row-value-cell.component.ts:32]: `ngSwitch` doesn't play nice with primitives yet, so we have to make this rather redundant looking switch statement. On the plus side, it keeps the template a bit cleaner.

File: web/src/app/pages/dashboard/transactions/transaction-details-dialog/transaction-details-dialog.component.ts [hunk 4]
```diff
@@ -3,6 +3,8 @@ import { MatDialogRef, MAT_DIALOG_DATA } from '@angular/material';
 import { EncryptedMessage, Message, Account } from 'src/app/lib/model';
 import { CryptoService, BurstService, StoreService } from 'src/app/lib/services';
 
+type TransactionDetailsCellValue = string | Message | EncryptedMessage | number;
```
Comment (ohager) [inline web/src/app/pages/dashboard/transactions/transaction-details-dialog/transaction-details-dialog.component.ts:6]: like that! I need to get more into the TS stuff.

File: web/locales/cs.json [hunk 5]
```diff
@@ -662,7 +662,7 @@
    "raw_transaction_details" : "Raw Transaction Details",
    "recent_blocks" : "Recent Blocks",
    "recent_transactions" : "Poslední transakce",
-   "recipient" : "Příjemce",
+   "recipientAddress" : "Příjemce",
```
Comment (ohager) [inline web/locales/cs.json:665]: recipient_address would be more consistent

File: web/src/app/lib/services/__tests__/burst.service.spec.ts [hunk 6]
```diff
@@ -16,7 +16,7 @@ describe('BurstService', () => {
 
         const storeServiceMock = new StoreService(testConfigFactory());
         storeServiceMock.settings = new BehaviorSubject(new Settings());
-
+        storeServiceMock.getSelectedAccount = () => Promise.resolve(new Account())
```
Comment (ohager) [inline web/src/app/lib/services/__tests__/burst.service.spec.ts:19]: I still"owe you" the tests 🙄

File: web/src/app/lib/services/burst.service.ts [hunk 7]
```diff
@@ -258,4 +260,148 @@ export class BurstService {
             return nameKey.replace(/_/g, " ");
         }
     };
+
+
+    public getTransactionNameFromType(transaction: Transaction, account: Account) {
```
Comment (ohager) [inline web/src/app/lib/services/burst.service.ts:265]: Wow....this screams for refactoring...time will come.

File: web/locales/cs.json (hunk 5 above)
Comment (blankey1337) [reply web/locales/cs.json:665]: Agreed, unfortunately this is how it comes back from the service. This is why some cell values are translated and others are not. What we need is some sort of parser, a camelCaseToUnderlineLowercase function would probably do the trick for 90% of the cell values. I'l see about that in a follow-up PR.

File: web/src/app/lib/services/burst.service.ts (hunk 7 above)
Comment (blankey1337) [reply web/src/app/lib/services/burst.service.ts:265]: Yeah... 🙄

File: web/src/app/lib/services/burst.service.ts [hunk 8]
```diff
@@ -258,4 +260,148 @@ export class BurstService {
...
+                break;
+            }
+        }
+        else if (transaction.type == 22) {
+            switch (transaction.subtype) {
+            case 0:
+                transactionType = "AT Creation";
+                break;
+            case 1:
+                transactionType = "AT Payment";
```
Comment (blankey1337) [inline web/src/app/lib/services/burst.service.ts:400]: Lines 368-400 also need translation.

**Comment by ohager** [comment]
@spebern pls, merge it... :wink:

**Pull request closed without merging**